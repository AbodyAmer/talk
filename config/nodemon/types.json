{
  "exec": "npm run compile:server:types",
  "ext": "graphql",
  "watch": [
    "./src/core/server/graph"
  ]
}