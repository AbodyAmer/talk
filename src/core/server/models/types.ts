import { FilterQuery } from 'mongodb';
import { Writeable } from '../../common/types';

export type FilterQuery<T> = Writeable<Partial<T>> &
    FilterQuery<Writeable<Partial<T>>>;
