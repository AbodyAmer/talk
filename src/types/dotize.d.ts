declare module 'dotize' {
    export = dotize;

    function dotize(obj: any): { [_: string]: any };
}
