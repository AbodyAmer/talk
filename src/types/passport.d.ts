import { User } from "talk-server/models/user";

declare module "express" {
  interface Request {
    user?: User;
  }
}
