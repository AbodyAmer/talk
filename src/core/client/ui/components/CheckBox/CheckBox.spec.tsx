import React from "react";
import TestRenderer from "react-test-renderer";

import { PropTypesOf } from "talk-ui/types";

import CheckBox from "./CheckBox";

it("renders correctly", () => {
  const props: PropTypesOf<typeof CheckBox> = {
    className: "custom",
    id: "checkboxID",
    value: "true",
    children: "Yes I agree",
  };
  const renderer = TestRenderer.create(<CheckBox {...props} />);
  expect(renderer.toJSON()).toMatchSnapshot();
});
