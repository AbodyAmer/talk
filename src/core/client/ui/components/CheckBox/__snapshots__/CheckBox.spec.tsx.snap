// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Flex-root CheckBox-root custom Flex-flex Flex-alignCenter"
>
  <input
    className="CheckBox-input"
    id="checkboxID"
    onBlur={[Function]}
    onFocus={[Function]}
    type="checkbox"
    value="true"
  />
  <label
    className="CheckBox-label"
    htmlFor="checkboxID"
  >
    Yes I agree
  </label>
</div>
`;
